frontend/src/features/research/activities/list/ActivityResults/AddActivityForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { PlusButton } from 'components/common/buttons';
+import { Select, SelectOption } from 'components/common/form';
+import { Formik } from 'formik';
+import { useApiResearchFile } from 'hooks/pims-api/useApiResearchFile';
+import useIsMounted from 'hooks/useIsMounted';
+import { Api_Activity, Api_ActivityTemplate } from 'models/api/Activity';
+import React, { useCallback, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+
+export interface IAddActivityFormProps {
+  onAddActivity: (activity: Api_Activity) => void;
+}
+
+export interface IAddActivity {
+  activityTypeId?: string;
+}
+
+export function AddActivityForm(props: IAddActivityFormProps) {
+  const { onAddActivity } = props;
+  const isMounted = useIsMounted();
+  const { getActivityTemplates } = useApiResearchFile();
+  const [templateTypes, setTemplateTypes] = useState<SelectOption[]>([]);
+
+  const fetchActivityTemplates = useCallback(async () => {
+    // const data = mockActivitiesResponse();
+    // TODO Call actual api endpoint
+    await getActivityTemplates().then(response => {
+      let options = response.data.map(
+        (template: Api_ActivityTemplate) =>
+          ({
+            value: template.id,
+            label: template.activityTemplateTypeCode?.description,
+            code: template.activityTemplateTypeCode?.id,
+          } as SelectOption),
+      );
+      setTemplateTypes(options);
+      if (options && isMounted()) {
+        setTemplateTypes(options);
+      }
+    });
+  }, [getActivityTemplates, isMounted]);
+
+  React.useEffect(() => {
+    fetchActivityTemplates();
+  }, [fetchActivityTemplates]);
+
+  const addActivity = (values: IAddActivity) => {
+    console.log(values);
+    const activity = { activityTemplateId: values.activityTypeId } as Api_Activity;
```

Comment:
should not coerce the types. Specially here since it invalidates the type checking for the API call

Reply:
removed passing only activitytypeid now